gnocchi/chef.py

```diff
@@ -45,9 +45,103 @@ class Chef(object):
     def __init__(self, coord, incoming, index, storage):
         self.coord = coord
         self.incoming = incoming
+        # This variable is an instance of the indexer,
+        # which means, database connector.
         self.index = index
         self.storage = storage
 
+    def clean_raw_data_inactive_metrics(self):
+        """Cleans metrics raw data if they are inactive.
+
+        The truncating of the raw metrics data is only done when new
+        measures are pushed. Therefore, if no new measures are pushed, and the
+        archive policy was updated to reduce the backwindow, the raw
+        datapoints for metrics that are not receiving new datapoints are never
+        truncated.
+
+        The goal of this method is to identify metrics that are in
+        "inactive state", meaning, not receiving new datapoints, and execute
+        their raw data points truncation. We check the column
+        "needs_raw_data_truncation", to determine if the archive policy was
+        updated, and no measure push was executed for the metric.
+
+        If the metric is not receiving new datapoints, the processing workflow
+        will not mark the column "needs_raw_data_truncation" to False;
+        therefore, that is how we identify such metrics.
+        """
+
+        metrics_to_clean = self.index.list_metrics(
+            attribute_filter={"==": {
+                "needs_raw_data_truncation": True}}
+        )
```

Comment:
i just realised, technically there can be more than one janitor if you have multiple metricd services. this could lead to each janitor getting the racing over the same list here. i don't see any issues aside from wasted redundancy but you see any issues with running below logic multiple times on same metric?

Reply:
Yes, but if the lock is already acquired by other, the Janitor would just skip that one. What we can do is to random sort the list. What do you think? This would minimize the situation where Janitors would try to process the same. Then, inside the lock, we can check again if the metric still needs processing. What do you think?